### PR TITLE
Basic JPAC2-11 Support

### PR DESCRIPTION
**Pull request by elijah-thomas774**
As title says, basics support for JPAC2-11 (used by Skyward Sword). 
This still contains  a bug that stalls (reload restores it back to normal) on a small amount of resources (only a few) (around the 1600 or 1700 index into skyward swords particle file). This however makes it so most effects can be viewed as the format is still largely identical to JPAC2-10.

### CODE REVIEW

**Comment by magcius** [comment]
Hi, sorry for taking so long to get around to this. Given the similarity between JPAC2-10 and JPAC2-11, it might make sense to just handle it in `parseResource_JPAC2_10` -- make it accept a version parameter, and just adjust the parsing code for the handful of changed blocks depending on the version. I pushed a flag cleanup in 8aca3a867ff68e449da95e2cedc73149b4c37856 which should hopefully make it easier to understand and modify.

**Comment by elijah-thomas774** [comment]
Ive updated it with the flag cleanup, the status of how well it works is still the same prior

File: src/Common/JSYSTEM/JPA.ts [hunk 1]
```diff
@@ -4426,7 +4427,12 @@ function parseResource_JPAC1_00(res: JPAResourceRaw): JPAResource {
     };
 }
 
-function parseResource_JPAC2_10(res: JPAResourceRaw): JPAResource {
+function parseResource_JPAC2_10(res: JPAResourceRaw, version: number): JPAResource {
```
Comment (magcius) [inline src/Common/JSYSTEM/JPA.ts:4430]:
```ts
function parseResource_JPAC2_10(res: JPAResourceRaw, version: JPACVersion.JPAC2_10 | JPACVersion.JPAC2_11) {

File: src/Common/JSYSTEM/JPA.ts [hunk 2]
```diff
@@ -4531,15 +4537,36 @@ function parseResource_JPAC2_10(res: JPAResourceRaw): JPAResource {
...
-            // 23 = unk
-            const isEnableTexScrollAnm = !!((flags >>> 24) & 0x01);
-            const tilingS              = !!((flags >>> 25) & 0x01) ? 2.0 : 1.0;
-            const tilingT              = !!((flags >>> 26) & 0x01) ? 2.0 : 1.0;
-            const isNoDrawParent       = !!((flags >>> 27) & 0x01);
-            const isNoDrawChild        = !!((flags >>> 28) & 0x01);
+            let isEnableProjection, isDrawFwdAhead, isDrawPrntAhead, 
+                isEnableTexScrollAnm, tilingS, tilingT, isNoDrawParent, 
+                isNoDrawChild : boolean;
+            if (!isJPAC2_11) // JPAC2_10
```
Comment (magcius) [inline src/Common/JSYSTEM/JPA.ts:4543]: Please obey existing formatting -- braces on same line. This should also be `if (version === JPACVersion.JPAC2_10)` now.

File: src/Common/JSYSTEM/JPA.ts [hunk 3]
```diff
@@ -4769,22 +4798,24 @@ function parseResource_JPAC2_10(res: JPAResourceRaw): JPAResource {
...
+            const scale = Math.pow(2, view.getInt8(tableIdx + (!isJPAC2_11 ? 0x24 : 0x4D)));
             const indTextureMtx = new Float32Array([
                 p00*scale, p01*scale, p02*scale, scale,
                 p10*scale, p11*scale, p12*scale, 0.0,
             ]);
 
             const indTextureMode: IndTextureMode = ((flags >>> 0) & 0x01);
-            const indTextureID = view.getUint8(tableIdx + 0x25);
+            // JPAC2_11: Code also checks to make sure offset 0x4C != 0
+            const indTextureID = view.getUint8(tableIdx + (!isJPAC2_11 ? 0x25 : 0x4E));
```
Comment (magcius) [inline src/Common/JSYSTEM/JPA.ts:4816]: Is this a TODO or not?

File: src/Common/JSYSTEM/JPA.ts [hunk 4]
```diff
@@ -4907,7 +4938,9 @@ function parseResource(version: JPACVersion, resRaw: JPAResourceRaw): JPAResourc
     else if (version === JPACVersion.JPAC1_00)
         return parseResource_JPAC1_00(resRaw);
     else if (version === JPACVersion.JPAC2_10)
-        return parseResource_JPAC2_10(resRaw);
+        return parseResource_JPAC2_10(resRaw, 10);
+    else if (version === JPACVersion.JPAC2_11)
+        return parseResource_JPAC2_10(resRaw, 11);
```
Comment (magcius) [inline src/Common/JSYSTEM/JPA.ts:4943]:
```ts
else if (version === JPACVersion.JPAC2_10 || version === JPACVersion.JPAC2_11)
    return parseResource_JPAC2_10(resRaw, version);
```

**Review by magcius — CHANGES_REQUESTED** [review]
Sorry for the late review on this, I've been fairly busy.

**Pull request closed without merging**